biojava-core/src/main/java/org/biojava/nbio/core/sequence/GeneSequence.java

```diff
@@ -257,10 +257,11 @@ public ExonSequence removeExon(String accession) {
 	 * @param begin
 	 * @param end
 	 * @return exon sequence
+	 * @throws IllegalArgumentException if accessionID is already added.
 	 */
-	public ExonSequence addExon(AccessionID accession, int begin, int end) throws Exception {
+	public ExonSequence addExon(AccessionID accession, int begin, int end) {
 		if (exonSequenceHashMap.containsKey(accession.getID())) {
-			throw new Exception("Duplicate accesion id " + accession.getID());
+			throw new IllegalArgumentException("Duplicate accesion id " + accession.getID());
```

Comment:
Thanks for this change. This is a lot cleaner now. I think generally `throws Exception` signatures should be avoided